Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/ProtoFormatConverterExtensions.kt

```diff
@@ -9,18 +9,40 @@ object ProtoFormatConverterExtensions {
 
     private const val ROLLING_PERIOD = 144
     private const val DEFAULT_TRANSMISSION_RISK_LEVEL = 1
+    private const val TRANSMISSION_RISK_DAY_0 = 5
+    private const val TRANSMISSION_RISK_DAY_1 = 6
+    private const val TRANSMISSION_RISK_DAY_2 = 7
+    private const val TRANSMISSION_RISK_DAY_3 = 8
+    private const val TRANSMISSION_RISK_DAY_4 = 7
+    private const val TRANSMISSION_RISK_DAY_5 = 5
+    private const val TRANSMISSION_RISK_DAY_6 = 3
+    private const val TRANSMISSION_RISK_DAY_7 = 2
+    private val DEFAULT_TRANSMISSION_RISK_VECTOR = intArrayOf(
+        TRANSMISSION_RISK_DAY_0,
+        TRANSMISSION_RISK_DAY_1,
+        TRANSMISSION_RISK_DAY_2,
+        TRANSMISSION_RISK_DAY_3,
+        TRANSMISSION_RISK_DAY_4,
+        TRANSMISSION_RISK_DAY_5,
+        TRANSMISSION_RISK_DAY_6,
+        TRANSMISSION_RISK_DAY_7)
     private const val MAXIMUM_KEYS = 14
 
     fun List<TemporaryExposureKey>.limitKeyCount() =
-        this.sortedWith(compareBy({ it.rollingStartIntervalNumber })).asReversed().take(MAXIMUM_KEYS)
+        this.sortedWith(compareBy { it.rollingStartIntervalNumber }).asReversed().take(MAXIMUM_KEYS)
 
-    fun List<TemporaryExposureKey>.transformKeyHistoryToExternalFormat() = this.map {
-        KeyExportFormat.TemporaryExposureKey.newBuilder()
-            .setKeyData(ByteString.readFrom(it.keyData.inputStream()))
-            .setRollingStartIntervalNumber(it.rollingStartIntervalNumber)
-            .setRollingPeriod(ROLLING_PERIOD)
-            .setTransmissionRiskLevel(DEFAULT_TRANSMISSION_RISK_LEVEL)
-            .build()
+    fun List<TemporaryExposureKey>.transformKeyHistoryToExternalFormat() =
+        this.sortedWith(compareBy { it.rollingStartIntervalNumber })
+            .mapIndexed { index, it ->
+                val riskValue =
+                    if (index <= DEFAULT_TRANSMISSION_RISK_VECTOR.lastIndex) DEFAULT_TRANSMISSION_RISK_VECTOR[index]
```

Comment:
@tklingbeil the key of today starts with TRANSMISSION_RISK_DAY_1  and not with TRANSMISSION_RISK_DAY_0. there is an offset to be apply.

Reply:
You're right. I have added an offset of 1.